gdal/gcore/gdal.h

```diff
@@ -159,7 +159,8 @@ typedef enum
   */
 typedef struct
 {
-    /*! Version of structure (to allow future extensions of the structure) */
+    /*! Version of structure (to allow future extensions of the structure), 
```

Comment:
I'm wondering if the doc of the structure itself, at line 157, shoudn't be rather changed to suggest using INIT_RASTERIO_EXTRA_ARG(&sExtraArg)

Reply:
I just found it independently (and the hard way) :smile:

Reply:
Does resampling when writing works? Or is it only when reading?

Reply:
> Does resampling when writing works? Or is it only when reading?

only nearest resampling is supported on writing

Reply:
@rouault I just spent 2 hours trying to find the problem with resampling when writing... probably mention it too? :smile:

Reply:
> probably mention it too?

Yes, this is very valuable input! Perhaps in the doc of GDALDataset and GDALRasterBand IRasterIO() method ?

Reply:
@rouault , maybe in the extra structure because this is where you look. I don't know what is supposed to work? Simple nearest neighbor oversampling appears to work? Or it is not intentional?

Reply:
No, in fact it is just image replication, since I had a pattern it was creating the illusion of oversampling

Reply:
> No, in fact it is just image replication, since I had a pattern it was creating the illusion of oversampling

On the write side, nearest oversampling = image replication